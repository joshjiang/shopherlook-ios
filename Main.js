import React, { Component } from 'react'
import { View, StyleSheet, Image, ScrollView, Dimensions, FlatList, TouchableOpacity } from 'react-native'
import { SearchBar } from 'react-native-elements';
import Client from 'shopify-buy';
import * as base from './environment';
var Buffer = require('buffer/').Buffer;

// connect to shopify api 
const client = Client.buildClient({
  domain: 'shopherlook.myshopify.com/',
  storefrontAccessToken: base.SHOPIFY_ACCESS_TOKEN
});


class Main extends Component {
  state = {
    products: [],
  };

  static navigationOptions = {
    title: 'Discover',
  };


  // make call to api
  // store in local component
  componentDidMount() {
    client.product.fetchAll().then((res) => {
      this.setState({
        products: res,
      });
    });
  }


  render() {

    return (
      <View style={styles.container}>
        <DiscoverContainer products={this.state.products} navigation={this.props.navigation} />
      </View>
    )
  }
}

// ui container for shopify data and navigation
export const DiscoverContainer = ({ products, navigation }) =>
  <View>
    <Disc products={products} larry={"asfasdf"} navigation={navigation} />
  </View>


// discover formatting
const formatData = (data, numColumns) => {
  const numberOfFullRows = Math.floor(data.length / numColumns);

  let numberOfElementsLastRow = data.length - (numberOfFullRows * numColumns);
  while (numberOfElementsLastRow !== numColumns && numberOfElementsLastRow !== 0) {
    data.push({ key: `blank-${numberOfElementsLastRow}`, empty: true });
    numberOfElementsLastRow++;
  }

  return data;
};

const numColumns = 3;


export class Disc extends Component {

  constructor() {
    super();
    this.state = {
      search: '',
    }
  }

  // state update
  updateSearch = search => {
    this.setState({ search });
  };

  renderItem = ({ item }) => {
    if (item.empty === true) {
      return <View style={[styles.item, styles.itemInvisible]} />;
    }

    return (
      <View style={styles.item}>
        {item}
      </View>
    )
  }

  render() {
    let products = this.props.products;

    const { search } = this.state;

    products = products.filter(
      (product) => {
        return product.title.includes(search) || product.title.toLowerCase().includes(search);
      }
    )

    const listProducts = products.map((product) =>
      <Look product={product} key={product.title} navigation={this.props.navigation}></Look>
    )

    // search initation
    return (
      <View>
        <SearchBar
          placeholder="Search"
          lightTheme
          onChangeText={this.updateSearch}
          value={search}
        />

        <View style={{ height: 1100 }}>
          <ScrollView >
            <FlatList
              data={formatData(listProducts, numColumns)}
              style={styles.container}
              renderItem={this.renderItem}
              numColumns={numColumns}
            />
          </ScrollView>
        </View>

      </View>
    )
  }
}

class Look extends React.Component {
  constructor() {
    super();

    this.state = {
      name: '',
      handle: '',
      id: '',

    };
  }

  componentDidMount() {
    console.log('https://shopherlook-sell.app/API/profileByStoreID/?storeID=' + Buffer.from(this.props.product.id, 'base64').toString().split('/')[4]);

    return fetch('https://shopherlook-sell.app/API/profileByStoreID/?storeID=' + Buffer.from(this.props.product.id, 'base64').toString().split('/')[4])
      .then((response) => response.json())
      .then((responseJson) => {

        this.setState({
          name: responseJson.first_name + ' ' + responseJson.last_name,
          handle: responseJson.instagram_handle,
          id: responseJson.ID,
        });

      })
      .catch((error) => {
        console.error(error);
      });
  }

  render() {

    return (
      <TouchableOpacity onPress={() => this.props.navigation.navigate('SinglePostScreen', {
        productId: this.props.product.id,
        id: this.state.id, //product
        proID: Buffer.from(this.props.product.id, 'base64').toString().split('/')[4],
      })}>

        <LookPicture photo={this.props.product.images[0].src} />

      </TouchableOpacity>
    )
  }
}


// discover thumbnails
export const LookPicture = ({ photo }) =>
  <Image source={{ uri: photo }} resizeMode="contain" style={styles.lookPhoto} />


// stylesheet
const styles = StyleSheet.create({
  container: {
    flex: 1,

  },
  header: {
    justifyContent: 'space-between',
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: 5,
    marginBottom: 20,
    marginRight: 10,
    marginLeft: 10
  },

  filterStyle: {
    justifyContent: 'space-between',
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: 20,
    marginBottom: 20,
    marginRight: 100,
    marginLeft: 100
  },

  lookPhoto: {
    resizeMode: 'stretch',
    height: 125,
    width: 125
  },

  betweenLooks: {

    marginTop: 5,
    marginBottom: 30,
    marginLeft: 40,
    marginRight: 40
  },

  item: {
    backgroundColor: 'white',
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
    margin: 1,
    height: Dimensions.get('window').width / numColumns, // approximate a square
  },

  itemInvisible: {
    backgroundColor: 'transparent',
  },

});

export default Main;