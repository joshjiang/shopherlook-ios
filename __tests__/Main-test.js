import React from 'react';
import Main from '../Main';
import renderer from 'react-test-renderer';

/*
describe('<Feed />', () => {
    it('has 1 child', () => {
        const tree = renderer.create(<Feed />).toJSON();
        expect(tree.children.length).toBe(1);
    });
});

*/
test('renders correctly', () => {
    const tree = renderer.create(<Main />).toJSON();
    expect(tree).toMatchSnapshot();
});