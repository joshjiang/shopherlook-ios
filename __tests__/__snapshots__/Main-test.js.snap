// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#e1e8ee",
            "borderBottomColor": "#e1e1e1",
            "borderBottomWidth": 1,
            "borderTopColor": "#e1e1e1",
            "borderTopWidth": 1,
            "padding": 8,
          }
        }
      >
        <View
          style={
            Object {
              "paddingHorizontal": 0,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#bdc6cf",
                "borderBottomWidth": 0,
                "borderColor": "#86939e",
                "borderRadius": 3,
                "flexDirection": "row",
                "minHeight": 30,
                "overflow": "hidden",
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "marginLeft": 8,
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                  underlayColor="white"
                >
                  <Text />
                </View>
              </View>
            </View>
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onClear={[Function]}
              onFocus={[Function]}
              placeholder="Search"
              placeholderTextColor="#86939e"
              platform="default"
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#86939e",
                  "flex": 1,
                  "fontSize": 18,
                  "marginLeft": 10,
                  "minHeight": 40,
                }
              }
              testID="searchInput"
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "marginLeft": 15,
                  "marginRight": 8,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 1100,
          }
        }
      />
    </View>
  </View>
</View>
`;
